lc_macro_pipeline/macro_pipeline.py

```diff
@@ -78,12 +79,33 @@ def _run_task(f):
     #     bag = dask.bag.from_delayed(delayed_tasks)
     #     return bag.compute()
 
+    def setup_client(self, 
+                     mode = 'auto',
+                     address='0.0.0.0', 
+                     num_workers=1, 
+                     port=0, 
+                     num_threads_per_worker=1, 
+                     memory_limit=0,
+                     silence_logs=False):
```

Comment:
The following are just suggestions, let me know what you guys think. 
The initialisation of the cluster has many optional inputs, and they might differ depending on the type of cluster that the user wants to build. So maybe leaving the input open with `*args`/`**kwargs` and have one flag choosing which type of cluster to build?

```suggestion
    def setup_client(self, 
                     mode='local',
                     *args,
                     **kwargs):
```